src/main/java/io/deephaven/csv/util/GroupWaiter.java

```diff
@@ -0,0 +1,90 @@
+package io.deephaven.csv.util;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+
+/**
+ * This simple class submits tasks to an {@link ExecutorService} and allows the caller to wait until (1) all of them
+ * complete, (2) any of them throws, whichever happens first.
+ */
+public class GroupWaiter {
```

Comment:
`java.util.concurrent.ExecutorCompletionService` may be a more relevant built-in that you can rely on instead of rolling our own?

Reply:
Ah, that's great, thank you! Please take another look.